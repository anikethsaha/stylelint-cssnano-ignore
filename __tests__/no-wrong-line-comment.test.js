const run = require('./helper/runplugin');

it('should throw error for adding ignore comment at selector', (done) => {
  run(`/* postcss-ignore-line */
  ul li {
    padding: 5px;
  }`).then((res) => {
    expect(res[0].text).toMatchSnapshot();
    done();
  });
});

it('should not throw any error message if comment used over decl', (done) => {
  run(`
  ul li {
    /* postcss-ignore-line */
    padding: 5px;
  }`).then((res) => {
    expect(res.length).toBe(0);
    done();
  });
});

it('should throw error for adding ignore comment at atRule', (done) => {
  run(`
  /* postcss-ignore-line */
 @media only screen and (max-width: 600px) {
  body {
    background-color: lightblue;
  }
}
  `).then((res) => {
    expect(res[0].text).toMatchSnapshot();
    done();
  });
});
