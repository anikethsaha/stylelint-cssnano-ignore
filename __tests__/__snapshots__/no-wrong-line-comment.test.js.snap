// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should throw error for adding ignore comment at atRule 1`] = `"cssnano-ignore-line comment written for wrong node type 'atrule' i.e 'media' at line 3:2 (cssnano-ignore-plugin/no-wrong-line-comment)"`;

exports[`should throw error for adding ignore comment at selector 1`] = `"cssnano-ignore-line comment written for wrong node type 'rule' i.e 'ul li' at line 2:3 (cssnano-ignore-plugin/no-wrong-line-comment)"`;
